client/components/app.jsx

```diff
@@ -5,6 +5,16 @@ class App extends React.Component {
   constructor(prop) {
     super(prop);
     this.state = { grades: [] };
+    this.getAverageGrade = this.getAverageGrade.bind(this);
```

Comment:
You don't need to bind this since you are not passing it anywhere as a callback.